src/features/device-heartbeat/index.ts

```diff
@@ -254,10 +254,13 @@ export class DeviceOnlineStateManager extends EventEmitter<{
 				id: deviceId,
 				options: {
 					$filter: {
-						$not: body,
+						$not: baseBody,
 					},
 				},
-				body,
+				body: {
+					...baseBody,
+					last_api_heartbeat_state_change_event: Date.now(),
```

Comment:
I originally added this as a hook, but at the end I preferred moving it here
* so that we can avoid the extra `.get()` request to confirm that the value has indeed changed, or blindly setting `last_api_heartbeat_state_change_event` whenever `values.api_heartbeat_state != null`.
* since this is the only place that we should be updating the heartbeat anyway.

Let me know though if you prefer to switch it back to hooks.